File: Electronics_Shop_17.Model/Requests/AddNotification.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddNotification
    {
        public string Title { get; set; }
        public string Message { get; set; }
        public DateTime DateCreated { get; set; } = DateTime.UtcNow;
        public bool IsGeneral { get; set; }
    }
}

File: angular17_web/src/app/models/notification/add.notification.ts
export class AddNotification {
  title!: string;
  message!: string;
  dateCreated?: Date;
  isGeneral!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.title = data.title;
    this.message = data.message;
    this.dateCreated = data.dateCreated ? new Date(data.dateCreated) : undefined;
    this.isGeneral = data.isGeneral;
  }
}
